### ISSUE TITLE
CUDA memory  error

### ISSUE THREAD

**Issue by HRain1016** [JDAI-CV/DCL#12]
I have two 1080ti GPU and ech of their memory is 11G. But when I train DCL on kaggle product dataset,  I only can set train batch 4 and val batch 16. Though in this way, GPU is used more than 90%.  So may I get your GPUs info and some suggestion?

**Comment by Chen94yue** [comment]
There is no good solution to this problem. We have at least 4 p40 GPUS with total 88G memory.

**Comment by akira-l** [comment]
Maybe you can try:
1. lower resolution, though hurt performance
2. use gradient accumulation to get a larger batch size
3. duly use empty_cache or use other methods to manage memory finely
4. use pytorch checkpoint to run partial model at once 
but the best solution is stilll getting more powerful computing resources if avaliable. (-_-|||)

**Comment by HRain1016** [comment]
> 
> 
> Maybe you can try:
> 
>     1. lower resolution, though hurt performance
> 
>     2. use gradient accumulation to get a larger batch size
> 
>     3. duly use empty_cache or use other methods to manage memory finely
> 
>     4. use pytorch checkpoint to run partial model at once
> 
> 
> but the best solution is stilll getting more powerful computing resources if avaliable. (-_-|||)

OK, thans!

**Comment by HRain1016** [comment]
> 
> 
> There is no good solution to this problem. We have at least 4 p40 GPUS with total 88G memory.

Thank you!
